scipy/stats/tests/test_resampling.py

```diff
@@ -1393,6 +1393,29 @@ def test_batch_generator(self, iterable, batch, expected):
         got = list(_resampling._batch_generator(iterable, batch))
         assert got == expected
 
+    def test_finite_precision_statistic(self):
+        # Some statistics return numerically distinct values when the values
+        # should be equal in theory. Test that `permutation_test` accounts
+        # for this in some way.
+        x = [1, 2, 4, 3]
+        y = [2, 4, 6, 8]
+
+        def statistic(x, y):
+            return stats.pearsonr(x, y)[0]
+
+        res = stats.permutation_test((x, y), statistic, vectorized=False,
+                                     permutation_type='pairings')
+        r, pvalue, null = res.statistic, res.pvalue, res.null_distribution
+
+        correct_p = 2 * np.sum(null >= r / (1+1e-14)) / len(null)
```

Comment:
to align the test case better to the implementation, should this be changed to `sum(null >= r - 1e-14)`?

Reply:
Yes, we'd want to do that.